pkgs/_macro_tool/lib/macro_runner.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'source_file.dart';
+
+/// Runs macros.
+abstract interface class MacroRunner {
+  /// Source files in the workspace.
+  List<SourceFile> get sourceFiles;
+
+  /// Runs macros in all files in the workspace, [sourceFiles].
+  ///
+  /// Returns augmentations and errors for each file.
+  ///
+  /// [injectImplementation] controls whether the `dart_model` macro
+  /// implementation is injected.
+  Future<WorkspaceResult> run({bool injectImplementation = true});
```

Comment:
Might want to rename `injectImplentation` to `runMacros` or similar. Really it is controlling whether macros are ran at all, and that it does that by injecting an implementation is more of a detail